aws-ts-static-website/index.ts

```diff
@@ -0,0 +1,210 @@
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+import { Output } from "@pulumi/pulumi";
+
+import * as fs from "fs";
+import * as mime from "mime";
+import * as path from "path";
+
+// Load the Pulumi program configuration. These act as the "parameters" to the Pulumi program,
+// so that different Pulumi Stacks can be brought up using the same code.
+
+const stackConfig = new pulumi.Config("static-website");
+const config = {
+    // pathToWebsiteContents is a relativepath to the website's contents.
+    pathToWebsiteContents: stackConfig.require("pathToWebsiteContents"),
+    // targetDomain is the domain/host to serve content at.
+    targetDomain: stackConfig.require("targetDomain"),
+    // ACM certificate for the target domain. Must be in the us-east-1 region.
+    certificateArn: stackConfig.require("certificateArn"),
+};
+
+// contentBucket is the S3 bucket that the website's contents will be stored in.
+const contentBucket = new aws.s3.Bucket(
+    "contentBucket",
+    {
+        bucket: config.targetDomain,
+        acl: "public-read",
+        // Configure S3 serve bucket contents as a website. This way S3 will automatically convert
+        // requests for "foo/" to "foo/index.html".
+        website: {
+            indexDocument: "index.html",
+            errorDocument: "404.html",
+        }
+    });
+
+// crawlDirectory recursive crawls the provided directory, applying the provided function
+// to every file it contains. Doesn't handle cycles from symlinks.
+function crawlDirectory(dir: string, f: (_: string) => void) {
+    const files = fs.readdirSync(dir);
+    for (const file of files) {
+        const filePath = `${dir}/${file}`;
+        const stat = fs.statSync(filePath);
+        if (stat.isDirectory()) {
+            crawlDirectory(filePath, f);
+        }
+        if (stat.isFile()) {
+            f(filePath);
+        }
+    }
+}
+
+// Sync the contents of the source directory with the S3 bucket, which will in-turn show up on the CDN.
+const webContentsRootPath = path.join(process.cwd(), config.pathToWebsiteContents);
+console.log("Syncing contents from local disk at", webContentsRootPath);
+crawlDirectory(
+    webContentsRootPath,
+    (filePath: string) => {
+        const relativeFilePath = filePath.replace(webContentsRootPath + "/", "");
+        // Create the aws.S3.BucketObject for every file. This has Pulumi track every file change for each
+        // update to the website. However, for large website it may be more efficient to not manage individual
+        // files using pulumi and instead just use the AWS CLI to sync files instead. (e.g. `aws s3 sync`.)
```

Comment:
Perhaps move this comment into the README?

Reply:
Done.